### PR TITLE
Add basic custom healthcheck service

### PR DESCRIPTION
**Pull request by toplenboren**
{"redis":"pass","postgres":"pass"}

Check Reids and Postgres in any time

### CODE REVIEW

**Review by sitozzz — APPROVED** [review]

File: packages/keystone/healthCheck.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            this.keystone = keystone
+        },
+        run: async () => {
+            try {
+                const res = await this.keystone.adapter.knex.raw('SELECT 1')
+                if (res) { return true }
+            } catch (e) { return false }
+        },
+    }
+}
```
Comment (sitozzz) [inline packages/keystone/healthCheck.js:113]: Maybe convert this function to a class? It looks very similar to another check

File: packages/keystone/healthCheck.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        name: 'redis',
+        prepare: () => {
+            this.redisClient = getRedisClient(clientName)
+        },
+        run: async () => {
+            try {
+                const client = this.redisClient
+                const res = await client.ping()
+                if (res === 'PONG') { return true }
+            } catch (e) { return false }
```
Comment (Donny1995) [inline packages/keystone/healthCheck.js:95]: making check function return optional error (success is when there is no error ) may make this thing move informative?

File: packages/keystone/healthCheck.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+const getPostgresHealthCheck = () => {
+    return {
+        name: 'postgres',
+        prepare: ({ keystone }) => {
+            this.keystone = keystone
+        },
+        run: async () => {
+            try {
+                const res = await this.keystone.adapter.knex.raw('SELECT 1')
+                if (res) { return true }
```
Comment (pahaz) [inline packages/keystone/healthCheck.js:109]: Do you want to check the result?

File: packages/keystone/healthCheck.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+const express = require('express')
+const LRUCache = require('lru-cache')
+
+const { getRedisClient } = require('./redis')
+
+const DEFAULT_HEALTHCHECK_URL = '/server-health'
+const DEFAULT_INTERVAL = 1000 // 1s
+const DEFAULT_TIMEOUT = 5000 // 5s
+const DELIMITER = ','
```
Comment (pahaz) [inline packages/keystone/healthCheck.js:76]: 😀 it’s more readable if you inline it

**Review by pahaz — APPROVED** [review]

**Pull request merged**